### PR TITLE
Support Hummingbird 2.x.x

### PR DESCRIPTION
**Pull request by Joannis**
Supercedes #8

### CODE REVIEW

File: Sources/HBLambdaTest/maths.swift
Comment (adam-fowler) [inline Sources/HBLambdaTest/maths.swift:59]: It's a bit frustrating you have to write both `init` functions. The only way I can think of getting around this would be have a base protocol `_HBBaseRequestContext` that requires access to the  `coreContext` but with no `init` function requirement. Then setup `HBRequestContext` to conform to `_HBBaseRequestContext` and add the `init` from `Channel` requirement. And then have `HBLambdaRequestContext` that conforms to `_HBBaseRequestContext` and adds a `init` from lambda event requirement.

File: Sources/HummingbirdLambda/HBLambdaRequestContext.swift
Comment (adam-fowler) [inline Sources/HummingbirdLambda/HBLambdaRequestContext.swift:7]: Is it worthwhile adding a `HBBasicLambdaRequestContext` in a similar way to the way we have `HBBasicRequestContext`? Just to get people up and running.

**Review by adam-fowler — APPROVED** [review]
Looks good. Can we have some comments though.

File: Sources/HBLambdaTest/maths.swift
Comment (adam-fowler) [reply Sources/HBLambdaTest/maths.swift:59]: What do you think should we implement that before merging this? We could get rid of the RequestContextSource as we are have full control over the parameters being sent to the init function

File: Sources/HummingbirdLambda/Lambda.swift
Comment (adam-fowler) [inline Sources/HummingbirdLambda/Lambda.swift:65]: Can we make the assumption that bodies will be encoded in JSON?

File: Sources/HummingbirdLambda/HBLambdaRequestContext.swift
Comment (Joannis) [reply Sources/HummingbirdLambda/HBLambdaRequestContext.swift:7]: I agree, but we can't fill in what Event they're using

File: Sources/HBLambdaTest/maths.swift
Comment (Joannis) [reply Sources/HBLambdaTest/maths.swift:59]: That makes sense to me, glad I presented the code as-is to you because that's a great solution. Let's make a HB PR for that first

File: Sources/HummingbirdLambda/HBLambdaRequestContext.swift
Comment (Joannis) [reply Sources/HummingbirdLambda/HBLambdaRequestContext.swift:7]: I'll add a generic `HBBasicLambdaRequestContext<Event>`,

Comment (adam-fowler) [reply Sources/HummingbirdLambda/HBLambdaRequestContext.swift:7]: Can we not use `HBBasicLambdaRequestContext<Event>`?

File: Sources/HummingbirdLambda/Lambda.swift
Comment (Joannis) [reply Sources/HummingbirdLambda/Lambda.swift:65]: Not always, you're right.

File: Sources/HBLambdaTest/maths.swift
Comment (adam-fowler) [reply Sources/HBLambdaTest/maths.swift:59]: Ok I'll set something up in HB

File: Sources/HummingbirdLambda/HBLambdaRequestContext.swift
Comment (Joannis) [reply Sources/HummingbirdLambda/HBLambdaRequestContext.swift:7]: Hah, yeah that's what I wrote as well. But just wanted to make sure that you were aware we can't completely make it ready off the bat

**Comment by Joannis** [comment]
@adam-fowler is this PR good to merge?

**Comment by Joannis** [comment]
I do still lean on your hummingbird PR, we can wait on merging that, or I can make a new PR when it's merged

File: Sources/HummingbirdLambda/HBLambdaRequestContext.swift
Comment (adam-fowler) [inline Sources/HummingbirdLambda/HBLambdaRequestContext.swift:9]: Why don't you just pass in the whole `LambdaContext`. Maybe someone else will want to extract more information from it

File: Sources/HBLambdaTest/maths.swift
Comment (adam-fowler) [inline Sources/HBLambdaTest/maths.swift:53]: You don't need to store the router here. You only need to set the responder at `init`. Currently you are building the responder every time a request comes through

**Review by adam-fowler — CHANGES_REQUESTED** [review]
A couple of minor changes and we're good

File: Sources/HummingbirdLambda/HBLambdaRequestContext.swift
Comment (Joannis) [reply Sources/HummingbirdLambda/HBLambdaRequestContext.swift:9]: No particular reason actually

File: Sources/HBLambdaTest/maths.swift
Comment (Joannis) [reply Sources/HBLambdaTest/maths.swift:53]: That's not what happens, because we store the responder in the LambdaHandler.

Comment (Joannis) [reply Sources/HBLambdaTest/maths.swift:53]: The problem arises because there's no public type or generic type that represents the Responder of an HBRouterBuilder. Therefore, I cannot annotate the type of responder here. So in order to prevent building a new responder for each request, I made sure to cache the value in HBLambdaHandler.

Even if we remove `some HBResponder<Context>` from `HBRouterBuilder` by making `HBRouter` public - I think this is still a usecase that will pop up more often. In particular when your new router builder PR is merged.

**Review by adam-fowler — APPROVED** [review]

**Pull request merged**
